test/Fund.test.js

```diff
@@ -0,0 +1,234 @@
+    await expectRevert.unspecified(
+      Fund.new(constants.ZERO_ADDRESS, this.endaomentAdmin.address, {
+        from: admin,
+      })
+    );
+  });
+
+  it("denies invalid non-ADMIN wallet to construct fund contract", async function() {
+    await expectRevert.unspecified(
+      Fund.new(manager, this.endaomentAdmin.address, { from: manager })
+    );
+  });
+
+  it("creates funds with correct manager", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+    assert.isDefined(this.fund.address);
+  });
+
+  it("allows ADMIN to change manager", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+
+    await fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+      from: admin,
+    });
+    const new_manager = await fund_contract.manager();
+
+    await assert.equal(new_manager, newManager);
+  });
+
+  it("allows REVIEWER to change manager, only if not paused", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+
+    await fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+      from: reviewer,
+    });
+    const new_manager = await fund_contract.manager();
+
+    await assert.equal(new_manager, newManager);
+
+    await this.endaomentAdmin.pause(3, { from: pauser });
+
+    await expectRevert.unspecified(
+      fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+        from: reviewer,
+      })
+    );
+  });
+
+  it("denies non-ADMIN, non-REVIEWER from changing manager", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+
+    await expectRevert.unspecified(
+      fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+        from: accountant,
+      })
+    );
+  });
+
+  it("checks the recipeint of a grant with the OrgFactory", async function() {
+    const orgFactory = await OrgFactory.new(this.endaomentAdmin.address, {
+      from: admin,
+    });
+    await this.endaomentAdmin.setRole(5, orgFactory.address, {
+      from: admin,
+    });
+
+    const org = await orgFactory.createOrg(
+      123456789,
+      this.endaomentAdmin.address,
+      { from: accountant }
+    );
+
+    const orgChecked = await this.fund.checkRecipient(
+      org.logs[0].args.newAddress,
+      orgFactory.address,
+      { from: admin }
+    );
+
+    assert.isTrue(orgChecked);
+
+    const failingOrg = await this.fund.checkRecipient(
+      constants.ZERO_ADDRESS,
+      orgFactory.address,
+      { from: admin }
+    );
+
+    assert.isFalse(failingOrg);
+  });
```

Comment:
Complex test: Add one-line comments within test to describe what is going on at each step.